pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2565,10 +2612,73 @@ func (c *Cloud) getTaggedSecurityGroups() (map[string]*ec2.SecurityGroup, error)
 	return m, nil
 }
 
+// Almost identical to updateInstanceSecurityGroupsForLoadBalancer, but it only adds ssg rules to
+// instances' inbound rules.
+func (s *AWSCloud) updateInstanceSharedSecurityGroups(ssgID string, allInstances []*ec2.Instance) error {
```

Comment:
@justinsb If I had to reuse `updateInstanceSecurityGroupsForLoadBalancer` I would have to make some changes to `updateInstanceSecurityGroupsForLoadBalancer` which I tried to avoid as much as possible. One nice thing about having completely independent method is that we could make a unit test to verify the new function's functionality independently.  What do you think?